src/http/client.cc

```diff
@@ -195,7 +195,10 @@ void connection::shutdown() noexcept {
 }
 
 future<> connection::close() {
-    return when_all(_read_buf.close(), _write_buf.close()).discard_result().then([this] {
+    // #2661. At least output stream close can fail with exception, because
+    // the stream will do a flush. If connection never managed to send data, we
+    // will throw again here. Need to suppress these exceptions.
+    return when_all(_read_buf.close().handle_exception([](auto&&){}), _write_buf.close().handle_exception([](auto&&){})).discard_result().then([this] {
```

Comment:
IIUC, the race condition occurs when `close()` is called while make_request() is still in progress.


Specifically:
1. make_request() isn't fully resolved until _write_buf is completely flushed
2. If close() is called before this flush completes, we'll get exceptions from the streams

I think we might need to make it the caller's responsibility to await the future from `make_request()` before calling `close()` instead of silently discarding exceptions. this could hide potential bugs in the Seastar applications which either discard the future or close the connection without waiting for it.